backend/user_error.ts

```diff
@@ -1,5 +0,0 @@
-export class UserError extends Error {
```

Comment:
I removed the concept of `UserError`.

Working with these backends is already subtle and having this extra error mode makes it more so.

This means that the system is going to bias toward failing in the stuck direction, which also isn't good.

I'm not sure what the solution is, but opting for simplicity for now.